ocaml/xapi/xapi_guest_agent.ml

```diff
@@ -81,18 +81,32 @@ let networks path (list: string -> string list) =
     else
       [ipv4]
   in
-  (* Find all "ethn" or "xenbrn" under a path. *)
-  let find_eths path = List.fold_left
+  (* Find all "ethn", "xenbrn" or newer interface standard names
```

Comment:
A more general comment. I believe the code works but wonder whether a more natural formulation would be in terms of a pipeline of maps and filters. The code that splits an interface "eth5" into "eth" and "5" could be put into a local function like `split_after_prefix prefix:string -> string -> string * string`